homeassistant/helpers/update_coordinator.py

```diff
@@ -419,6 +419,29 @@ def async_set_updated_data(self, data: _DataT) -> None:
         self.async_update_listeners()
 
 
+class TimestampDataUpdateCoordinator(DataUpdateCoordinator[_DataT]):
+    """DataUpdateCoordinator which keeps track of the last successful update."""
+
+    last_update_success_time: datetime | None = None
```

Comment:
Is there any reason why this field couldn't be added to `DataUpdateCoordinator[_DataT]` and eliminate the need for this class?

Reply:
Things will get messy when another property is added that is used by some of each Coordinator type. Just my $0.02.

Reply:
Ok, and now it's not messy because the optional property is only added in a subclass?
Or maybe I misunderstand what you mean.

Reply:
I think adding subclasses like this is messy and it will get worse if done again.

Reply:
We've discussed this and think for now it's best with a separate coordinator as there's an additional function call that is relatively costly.

We can possibly refactor this in the future.